api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -176,22 +180,26 @@ public void setDistinctWorkspaceLimit(int limit) {
     // RW-1298
     // This needs to be refactored to only use namespace and FC ID
     // The purpose of this Map, is to check what is actually still present in FC
-    Map<Long, WorkspaceResponse> workspaceAccessMap =
-        workspaceIdList.stream()
-            .collect(
-                Collectors.toMap(
-                    id -> id,
-                    id -> {
-                      DbWorkspace workspace = workspaceService.findByWorkspaceId(id);
-                      WorkspaceResponse workspaceResponse =
-                          fireCloudService.getWorkspace(
-                              workspace.getWorkspaceNamespace(), workspace.getFirecloudName());
-                      return workspaceResponse;
-                    }));
+    ImmutableMap.Builder<Long, WorkspaceResponse> liveWorkspacesByIdBuilder = new Builder<>();
+    for (long workspaceId : workspaceIdList) {
+      final Optional<DbWorkspace> workspaceMaybe = workspaceService.getWorkspaceMaybe(workspaceId);
+      if (!workspaceMaybe.isPresent()) {
+        logger.log(
+            Level.INFO,
+            String.format("Workspace ID %d still in recent list but not found", workspaceId));
```

Comment:
But is it really something to be tolerated and expected?  I was certainly surprised it works this way, which is why I took so long on this PR.

`<rant>`
I still don't understand why we'd want to store archival/log information in our main application transaction database. (This is the only reason I've heard why we don't throw away unreachable URR entries or other workspace contents). I'd much rather have the relationships we want enforced by the RDBMS and spend programming time on building cool stuff instead of emulating the DB's constraint integrity and features in lots of places (but not everywhere).
`</rant>`

Obviously I'm in the minority, so I'll remove the statement.